docassemble/ALKilnInThePlayground/data/questions/run_alkiln_tests.yml

```
@@ -2159,83 +2552,428 @@ content: |
+
+    if not config_key in runtime_vals_data["result"]:
+      return {"ok": False, "message": "Could not find this key in the runtime config", "result": None }
+
+    return {
+      "ok": True, "message": "Found the value",
+      "result": runtime_vals_data["result"][ config_key ]
+    }
+---
+id: get config vals or ensure they're created
+need:
+  - get_config_path
+code: |
+  def get_json_from_file( json_filepath ):
+    """
+    Get values from the alkiln runtime config as json or, if 
+        they don't yet exist, make that json. Return info about
+        what happened.
+
+    Returns:
+      (dict):
+        ok (bool): True if succeeded, False if failed
+        message (str): If failed, text of what went wrong
+        result (dict): The json
+    """
+    import os
+    import json
+    import traceback
+
+    message = ""
+    result = {}
+
+    # config_path_result = get_config_path()
+    # if not config_path_result["ok"]:
+    #   return { "ok": False, "message": not config_path_result["message"], "result": result }
+    # local_config_path = config_path_result["result"]
+
+    try:
+      with open( json_filepath, 'r', encoding='utf-8') as json_file:
+        try:
+          result = json.load( json_file )
+      
+        except Exception as json_load_error:
+          message = f"""Error loading json from file at "{ json_filepath }":\n{ "\n".join(traceback.format_exception( json_load_error )) }"""
+          _log(f"""🔎 ALKiln WARNING ALKP0013: { message }""")
+          return {
+            "ok": False, "message": message, "result": result
+          }
+    except Exception as json_open_error:
+      message = f"""Skipped opening json file at "{ json_filepath }":\n{ "\n".join(traceback.format_exception( json_open_error )) }"""
+      _debug(f"""🖊️ ALKiln NOTE: { message }""")
+      return {
+        "ok": False, "message": message, "result": result
+      }
+
+        # try:
+        #   result = {}
+        #   json.dump(result, config_file, ensure_ascii=False, indent=2)
+  
+        # except Exception as config_dump_error:
+        #   message = f"""Error dumping json to config file at "{ local_config_path }":\n{ "\n".join(traceback.format_exception( config_dump_error )) }"""
+        #   _log(f"""🤕 ALKiln ERROR ALKP0015: { message }""")
+        #   return {
+        #     "ok": False, "message": message, "result": result
+        #   }
+  
+    return { "ok": True, "message": message, "result": result }
+---
+reconsider: True
+need:
+  - get_config_path
+  # - get_existing_path
+code: |
+  """
+  Return the runtime_config.json path or, if it doesn't yet exist, None
+  """
+  # temp_config_path = get_existing_path( "runtime_config.json" )
+  # if temp_config_path is None:
+  #   config_path_problem = 'No "runtime_config.json" file exists.'
+  #   _log(f'🤕 ALKiln ERROR: { config_path_problem }')
+  # config_path = temp_config_path
+  config_path = get_config_path()["result"]
```

We only use `config_path` when setting the folder name. Discuss: use `get_config_path()` when setting the folder name instead of `config_path` since `root_path` might change.

```suggestion
```

---

Alternatively, keep this as the default value:

```suggestion
reconsider: True
need:
  - get_config_path
code: |
  """
  Set the default path for runtime_config.json or, if it doesn't yet exist, None
  """
  config_path = get_config_path()["result"]
```